### PR TITLE
Include klib as KNOWN_ZIP_EXTENSIONS for java normalization

### LINKED ISSUE #21610: Include klib extension as KNOWN_ZIP_EXTENSIONS for java normalization
Projects using Kotlin Multiplatform generate `klib` binaries. Kotlin/Native libraries are zip files containing a predefined directory structure and should benefit from runtime classpath normalizations. 

https://kotlinlang.org/docs/native-libraries.html#library-format

### Expected Behavior
Gradle applies runtime classpath normalizations on klib extensions allowing cache hits on tasks using these extension outputs.

### Current Behavior
When comparing Build Scans, we observe: `Normalization: Runtime classpath, non-normalized line endings` differences on tasks consuming klibs causing execution of the tasks

### Context

When comparing builds investigating cache issues we observed:
Screen Shot 2022-08-22 at 3 49 55 PM

When we apply the diff on both files, we see only timestamp differences. Including this extension as KNOWN_ZIP_EXTENSIONS will help to retrieve task outputs from the cache

### PR DESCRIPTION
**Pull request by wolfs**
Projects using Kotlin Multiplatform generate `klib` binaries. Kotlin/Native libraries are zip files containing a predefined directory structure and should benefit from runtime classpath normalization. 

In projects using Kotlin MPP, when comparing builds for cache misses investigation we observe:
Screen Shot 2022-08-22 at 3 49 55 PM

This change allows applying classpath runtime normalization on tasks using klib binaries:
Screen Shot 2022-08-22 at 3 53 42 PM

To test the change, I used the project https://github.com/square/okio and applied [Build Validation scripts](https://github.com/gradle/gradle-enterprise-build-validation-scripts) (03-validate-local-build-caching-different-locations)

Fixes #21610 
Replaces #21611

### CODE REVIEW

File: subprojects/core/src/integTest/groovy/org/gradle/normalization/ConfigureRuntimeClasspathNormalizationIntegrationTest.groovy [hunk 1]
```diff
@@ -579,6 +579,95 @@ class ConfigureRuntimeClasspathNormalizationIntegrationTest extends AbstractInte
         FILE_FILTER_FLAG        | "ignore '**/ignored.txt'"                                                 | 'ignore rule'
     }
 
+    def "treats #extension as archive"() {
```
Comment (lptr) [inline subprojects/core/src/integTest/groovy/org/gradle/normalization/ConfigureRuntimeClasspathNormalizationIntegrationTest.groovy:582]:
```suggestion
    def "treats '#extension' as archive"() {
```

**Review by lptr — APPROVED** [review]
LGTM, left some nits.

**Comment by wolfs** [comment]
☝️ Unrelated flaky test: `finalizer can have dependencies that are not reachable from first discovered finalized task and reachable from second discovered finalized task`

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Pull request merged**